addons/stock_account/models/stock_quant.py

```diff
@@ -28,13 +37,10 @@ def _compute_value(self):
         """
         for quant in self:
             quant.currency_id = quant.company_id.currency_id
-            # If the user didn't enter a location yet while enconding a quant.
-            if not quant.location_id:
-                quant.value = 0
-                return
-
-            if not quant.location_id._should_be_valued() or\
-                    (quant.owner_id and quant.owner_id != quant.company_id.partner_id):
+            if not quant.location_id or not quant.product_id or\
+                    not quant.location_id._should_be_valued() or\
+                    quant._should_exclude_for_valuation() or\
+                    float_is_zero(quant.quantity, precision_rounding=quant.product_id.uom_id.rounding):
```

Comment:
```suggestion
            # If the user didn't enter a location yet while enconding a quant.
            if not quant.location_id:
                quant.value = 0
                return

            if not quant.location_id._should_be_valued() or\
                    quant._should_exclude_for_valuation():
```
As this is stable, let's just do what the initial commit did, which is only extract some conditions, instead of  rewriting them to match a future version. :slightly_smiling_face: